### PR TITLE
Fixes #20756 - make the activejob adapter Rails 5 compatible

### PR DESCRIPTION
**Pull request by iNecas**

### CODE REVIEW

**Comment by iNecas** [comment]
Tests should tell us enough if the thing works or not: we're running rails 4 for ruby < 2.2.0, the rails 5 for higher versions, so we're testing both cases.

File: lib/dynflow/active_job/queue_adapter.rb [hunk 1]
```diff
@@ -1,21 +1,26 @@
...
-          def enqueue(job)
-            ::Rails.application.dynflow.world.trigger(JobWrapper, job.serialize)
-          end
-
-          def enqueue_at(job, timestamp)
-            ::Rails.application.dynflow.world.delay(JobWrapper, { :start_at => Time.at(timestamp) }, job.serialize)
-          end
-        end
+        # For ActiveJob <= 4
+        extend QueueMethods
```
Comment (adamruzicka) [inline lib/dynflow/active_job/queue_adapter.rb]: Is it ok to have the methods defined both on the class and the instance? Wouldn't it be better to check `::ActiveJob.version` and act accordingly?

**Comment by adamruzicka** [comment]
Hmm, test failures on 2.0.0 are definitely not transient (`NoMethodError: private method 'include' called for ActiveJob::QueueAdapters:Module`)

**Comment by mmoll** [comment]
@iNecas any news here? :)

**Comment by iNecas** [comment]
Ok, it seems including the `Concerns` were causing the issues there (I can't think of any other reason for the issues with `include` method. Trying without it.

**Comment by iNecas** [comment]
@adamruzicka I've seen some more frequent intermittent failures on 

```
1) Failure:

action::Dynflow::Action::WithSubPlans::Dynflow::Action::WithPollingSubPlans#test_0001_polls for sub plans state [/home/travis/build/Dynflow/dynflow/test/action_test.rb:628]:

Expected: :planned

  Actual: :running

  2) Error:

action::Dynflow::Action::WithSubPlans::cancelling#test_0002_sends the abort event to all actions that are running and support cancelling:

fatal: No live threads left. Deadlock?

    /home/travis/.rvm/rubies/ruby-2.0.0-p648/lib/ruby/2.0.0/thread.rb:72:in `sleep'

    /home/travis/.rvm/rubies/ruby-2.0.0-p648/lib/ruby/2.0.0/thread.rb:72:in `block (2 levels) in wait'

    /home/travis/.rvm/rubies/ruby-2.0.0-p648/lib/ruby/2.0.0/thread.rb:68:in `handle_interrupt'

    /home/travis/.rvm/rubies/ruby-2.0.0-p648/lib/ruby/2.0.0/thread.rb:68:in `block in wait'

    /home/travis/.rvm/rubies/ruby-2.0.0-p648/lib/ruby/2.0.0/thread.rb:66:in `handle_interrupt'

    /home/travis/.rvm/rubies/ruby-2.0.0-p648/lib/ruby/2.0.0/thread.rb:66:in `wait'

    /home/travis/.rvm/gems/ruby-2.0.0-p648/gems/concurrent-ruby-edge-0.2.4/lib/concurrent/edge/future.rb:404:in `block in wait_until_complete'

    /home/travis/.rvm/gems/ruby-2.0.0-p648/gems/concurrent-ruby-edge-0.2.4/lib/concurrent/edge/future.rb:400:in `synchronize'

    /home/travis/.rvm/gems/ruby-2.0.0-p648/gems/concurrent-ruby-edge-0.2.4/lib/concurrent/edge/future.rb:400:in `wait_until_complete'

    /home/travis/.rvm/gems/ruby-2.0.0-p648/gems/concurrent-ruby-edge-0.2.4/lib/concurrent/edge/future.rb:586:in `value'

    /home/travis/build/Dynflow/dynflow/test/action_test.rb:610:in `block (4 levels) in <module:Dynflow>'

```

Mind looking (outside of scope for PR)

**Comment by iNecas** [comment]
@adamruzicka since the issues were pretty often (most of the sqlite tests were actually hitting this in travis), I gave a look at that as well in https://github.com/Dynflow/dynflow/pull/249/commits/f4e13dd39aedef4b814704180d70c1fdddf2eba6

**Comment by adamruzicka** [comment]
Thanks, I wanted to look at it tomorrow afternoon, looks like I won't have to.

**Comment by iNecas** [comment]
Ok, we got on standard 2 intermittent issues. Feel free to use spared time for hunting those :)

**Comment by mmoll** [comment]
💚

**Comment by adamruzicka** [comment]
Looks good to me, merging

**Comment by adamruzicka** [comment]
Thanks @iNecas !

**Comment by mmoll** [comment]
thanks, a new release soon would be nice to have this available in the released gem 💎

**Comment by iNecas** [comment]
Dynflow 0.8.29 just released

**Pull request merged**